models/demos/falcon7b/tests/test_perplexity_falcon.py

```diff
@@ -161,10 +182,43 @@ def run_test_perplexity(
     logger.info("Falcon Perplexity Check Passed!")
 
 
+@pytest.mark.parametrize(
+    "llm_mode, batch_size, max_seq_len, num_samples, expected_ppl",
+    (
+        ("prefill", 32, 1024, 64, 11.5),
+        ("decode", 64, 1024, 64, 11.5),
+    ),
+    ids=[
+        "prefill_seq1024",
+        "decode_1024",
+    ],
+)
+def test_perplexity_huggingface(
+    llm_mode,
+    batch_size,
+    max_seq_len,
+    num_samples,  # Total number of prompts to evaluate (all if None)
+    expected_ppl,
+    model_location_generator,
+):
+    run_test_perplexity(
+        llm_mode,
+        batch_size,
+        max_seq_len,
+        None,
+        model_location_generator,
+        None,
+        None,
+        num_samples,
+        expected_ppl,
+        use_hf_model=True,
+    )
+
+
 @pytest.mark.parametrize(
     "llm_mode, batch_size, max_seq_len, model_config_str, num_samples, expected_ppl",
     (
-        ("prefill", 1, 1024, "BFLOAT16-DRAM", 128, 11.5),
+        ("prefill", 1, 1024, "BFLOAT16-DRAM", 64, 12.0),
```

Comment:
could we add 128 and 2k seq lengths as well?

Reply:
Yes, I will try to add that early next week.